File: code/08/java/TestRunner.java
/* TestRunner.java tests Temperature.java and outputs a chart
 * Created by: Mark Davis
 * Date: April 12, 2016
 ************************************************************/

import java.util.Scanner;

public class TestRunner {

public static void main(String[] args) {
	Scanner keyboard = new Scanner(System.in);	

	//get base temperature
	System.out.println("Enter values for base temperature");
	Temperature baseTemp = new Temperature('F', 30);
	baseTemp.readTemp();
	
	//get limit temperature
	System.out.println("Enter values for limit temperature");
	Temperature limitTemp = new Temperature('F', 30);
	limitTemp.readTemp();

	//get step value
	System.out.println("Enter step value: ");
	double stepValue = keyboard.nextDouble();

	while( baseTemp.lessThan(limitTemp) ) {
		
		System.out.print(baseTemp.printTemp() + " ");
		baseTemp.getCelsius();
		System.out.print(baseTemp.printTemp() + " ");
		baseTemp.getKelvin();
		System.out.print(baseTemp.printTemp() + "\n");
		baseTemp.getFahrenheit();
		baseTemp.raise(stepValue);
	}    
}
}

File: code/08/ruby/project/TestRunner.rb
#TestRunner.rb tests Temperature.rb and outputs chart
#
#Created by: Mark Davis
#CALVIN COLLEGE, CS 214, project08
#April 12, 2016
#####################################################

require "./Temperature"

def testTemp
	#get user input for base temperature
	baseTemp = Temperatures::Temperature.new(100.0, "F")
	puts "Enter values for base temperature"
	baseTemp.readTemp()

	#get user input for limit temperature
	limitTemp = Temperatures::Temperature.new(100.0, "F")
	puts "Enter values for limit temperature"
	limitTemp.readTemp()

	#get user input for step value
	puts "Enter step value: "
	stepValue = gets.chomp.to_f

	while ( baseTemp.lessThan(limitTemp) ) do
		print baseTemp.printTemp 
		print " "
		baseTemp.setCelsius
		print baseTemp.printTemp 
		print " "
		baseTemp.setKelvin
		print baseTemp.printTemp 
		puts " "
		baseTemp.setFahrenheit
		baseTemp.raise(stepValue)
	end 
end

testTemp
